detekt-api/src/main/kotlin/io/gitlab/arturbosch/detekt/api/internal/BaseConfig.kt

```diff
@@ -37,9 +37,18 @@ abstract class BaseConfig : HierarchicalConfig {
 
     protected open fun tryParseBasedOnDefault(result: String, defaultResult: Any): Any = when (defaultResult) {
         is Int -> result.toInt()
-        is Boolean -> result.toBoolean()
+        is Boolean ->
+            if (result in ALLOWED_BOOL_VALUES) {
+                result.toBoolean()
+            } else {
+                throw ClassCastException()
+            }
         is Double -> result.toDouble()
         is String -> result
         else -> throw ClassCastException()
     }
+
+    companion object {
+        val ALLOWED_BOOL_VALUES = setOf("true", "false")
+    }
 }
```

Comment:
```suggestion
}

private val ALLOWED_BOOL_VALUES = setOf("true", "false")
```

What do you think? I realised about this when I saw this implementation detail in the documentation.